src/Mvc/Mvc.Core/src/ModelBinding/Binders/ServicesModelBinder.cs

```diff
@@ -23,9 +25,14 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
         }
 
         var requestServices = bindingContext.HttpContext.RequestServices;
-        var model = requestServices.GetRequiredService(bindingContext.ModelType);
+        var model = IsOptionalParameter ?
+            requestServices.GetService(bindingContext.ModelType) :
+            requestServices.GetRequiredService(bindingContext.ModelType);
 
-        bindingContext.ValidationState.Add(model, new ValidationStateEntry() { SuppressValidation = true });
+        if (model != null)
+        {
+            bindingContext.ValidationState.Add(model, new ValidationStateEntry() { SuppressValidation = true });
+        }
```

Comment:
Why is this changing?

Reply:
`GetServices` return an `object?` and the ValidationStateDictionary expects an `object` as key

Reply:
If it gets assigned the default value, do we try and validate the model (we shouldn't since it's not bound from the request)?

Reply:
If the model is set to null, that means the parameter is not required and we will skip the validation here:
https://github.com/dotnet/aspnetcore/blob/15fa3ad10859abcc54e3ad5557dc928f6c94994d/src/Mvc/Mvc.Core/src/ModelBinding/Validation/ValidationVisitor.cs#L187

Also, in case anything else is set we will suppress the validation as before.